tests/test_main_loop.py

```diff
@@ -0,0 +1,47 @@
+import unittest
```

Comment:
Why is this a new file? This was here before, right?

Reply:
It was, no idea why it was touched as a new file